src/wazuh_qa_framework/generic_modules/tools/file_regex_monitor.py

```diff
@@ -13,12 +13,105 @@
+        pattern (str): Regex that is combined with the prefix to match with logs.
+        prefix (str): Prefix string that comes before the pattern used to match logs.
+        complete_pattern (str): Prefix and pattern join.
+        regex (re): Regex created with the prefix and pattern.
+        monitored_file (str): File path to monitor.
+        callback (function): Callback function that will be evaluated for each log line.
+        timeout (int): Max time to monitor and trigger the callback.
+    """
+    def __init__(self, description=None, pattern='.*', prefix='.*', timeout=1, monitored_file=None, callback=None):
+        self.validate_args(description, pattern, prefix, timeout, monitored_file, callback)
+
+        self.pattern = pattern
+        self.prefix = prefix
+        self.complete_pattern = pattern if prefix is None else fr'{prefix}{pattern}'
+        self.regex = re.compile(self.complete_pattern)
+        self.timeout = timeout
+        self.monitored_file = monitored_file
+        self.callback = callback if callback else self.get_default_callback()
+        self.description = description if description else self.get_default_description()
+
+    def __str__(self):
+        """String representation when using str function."""
+        return self.description
+
+    def validate_args(self, description, pattern, prefix, timeout, monitored_file, callback):
+        """Validate the given monitoring args.
+
+        Args:
+            description (str): String that describes the whole metadata stack.
+            pattern (str): Regex that is combined with the prefix to match with logs.
+            prefix (str): Prefix string that comes before the pattern used to match logs.
+            monitored_file (str): File path to monitor.
+            callback (function): Callback function that will be evaluated for each log line.
+            timeout (int): Max time to monitor and trigger the callback.
+        """
+        # Validate if the specified file can be monitored.
+        # Check if monitored file is given
+        if not monitored_file:
+            raise ValidationError('The monitored_file arg is required. Pass an existing file to be monitored.')
+
+        # Check that the monitored file exists
+        if not os.path.exists(monitored_file):
+            raise ValidationError(f"File {monitored_file} does not exist")
+
+        # Check that the monitored file is a file
+        if not os.path.isfile(monitored_file):
+            raise ValidationError(f"{monitored_file} is not a file")
+
+        # Check that the program can read the content of the file
+        if not os.access(monitored_file, os.R_OK):
+            raise ValidationError(f"{monitored_file} is not readable")
+
+        # Validate if timeout is a positive integer
+        if timeout < 0:
+            raise ValidationError('The timeout can\'t be a negative value')
+
+        # Validate that callback is a callable function if given
+        if callback:
```

Comment:
You can simplify these two if in one:
```suggestion
        if callback and not callable(callback):
```

Reply:
Done https://github.com/wazuh/qa-system-framework/pull/58/commits/c8eae8375bb869545bb6daaa3650e4080c3bb741